hardware_interface/test/fake_components/test_generic_system.cpp

```diff
@@ -1027,3 +1100,182 @@ TEST_F(TestGenericSystem, generic_system_2dof_functionality_with_offset_custom_i
+  hardware_interface::LoanedStateInterface gpio2_vac_s =
+    rm.claim_state_interface("flange_vacuum/vacuum");
+  hardware_interface::LoanedCommandInterface gpio1_a_o1_c =
+    rm.claim_command_interface("flange_analog_IOs/analog_output1");
+  hardware_interface::LoanedCommandInterface gpio2_vac_c =
+    rm.claim_command_interface("flange_vacuum/vacuum");
+
+  // State interfaces without initial value are set to 0
+  ASSERT_TRUE(std::isnan(gpio1_a_o1_s.get_value()));
+  ASSERT_TRUE(std::isnan(gpio2_vac_s.get_value()));
+  ASSERT_TRUE(std::isnan(gpio1_a_o1_c.get_value()));
+  ASSERT_TRUE(std::isnan(gpio2_vac_c.get_value()));
+
+  // set some new values in commands
+  gpio1_a_o1_c.set_value(0.111);
+  gpio2_vac_c.set_value(0.222);
+
+  // State values should not be changed
+  ASSERT_TRUE(std::isnan(gpio1_a_o1_s.get_value()));
+  ASSERT_TRUE(std::isnan(gpio2_vac_s.get_value()));
+  ASSERT_EQ(0.111, gpio1_a_o1_c.get_value());
+  ASSERT_EQ(0.222, gpio2_vac_c.get_value());
+
+  // write() does not change values
+  rm.write();
+  ASSERT_TRUE(std::isnan(gpio1_a_o1_s.get_value()));
+  ASSERT_TRUE(std::isnan(gpio2_vac_s.get_value()));
+  ASSERT_EQ(0.111, gpio1_a_o1_c.get_value());
+  ASSERT_EQ(0.222, gpio2_vac_c.get_value());
+
+  // read() mirrors commands + offset to states
+  rm.read();
+  ASSERT_EQ(0.111, gpio1_a_o1_s.get_value());
+  ASSERT_EQ(0.222, gpio2_vac_s.get_value());
+  ASSERT_EQ(0.111, gpio1_a_o1_c.get_value());
+  ASSERT_EQ(0.222, gpio2_vac_c.get_value());
+
+  // set some new values in commands
+  gpio1_a_o1_c.set_value(0.333);
+  gpio2_vac_c.set_value(0.444);
+
+  // state values should not be changed
+  ASSERT_EQ(0.111, gpio1_a_o1_s.get_value());
+  ASSERT_EQ(0.222, gpio2_vac_s.get_value());
+  ASSERT_EQ(0.333, gpio1_a_o1_c.get_value());
+  ASSERT_EQ(0.444, gpio2_vac_c.get_value());
+
+  rm.start_components();
+  states_map = rm.get_components_states();
+  EXPECT_EQ(
+    states_map["GenericSystem2dof"].label(), hardware_interface::lifecycle_state_names::ACTIVE);
+
+  rm.stop_components();
+  states_map = rm.get_components_states();
+  EXPECT_EQ(
+    states_map["GenericSystem2dof"].label(), hardware_interface::lifecycle_state_names::INACTIVE);
+
+  // check other functionalities are working well
+  generic_system_functional_test(urdf);
+}
+
+TEST_F(TestGenericSystem, valid_urdf_ros2_control_system_robot_with_gpio_fake_command_)
+{
+  auto urdf = ros2_control_test_assets::urdf_head +
+              valid_urdf_ros2_control_system_robot_with_gpio_fake_command_ +
+              ros2_control_test_assets::urdf_tail;
+  hardware_interface::ResourceManager rm(urdf);
+
+  // check is hardware is configured
+  std::unordered_map<std::string, rclcpp_lifecycle::State> states_map;
+  states_map = rm.get_components_states();
+  EXPECT_EQ(
+    states_map["GenericSystem2dof"].label(), hardware_interface::lifecycle_state_names::INACTIVE);
+
+  ASSERT_EQ(8, rm.state_interface_keys().size());
+  ASSERT_EQ(8, rm.command_interface_keys().size());
+
```

Comment:
This is not needed. It is checked in the following lines.

```suggestion
```

Reply:
Done.